gridstatus/ercot.py

```diff
@@ -3351,3 +3353,64 @@ def _weather_zone_column_name_order(self):
             "Southern",
             "West",
         ]
+
+    @support_date_range(frequency=None)
+    def get_indicative_lmp_by_settlement_point(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        if date == "latest":
+            self.get_indicative_lmp_by_settlement_point(date="today")
+        if not end:
+            end = date + pd.DateOffset(days=1)
+
+        docs = self._get_documents(
+            report_type_id=ERCOT_INDICATIVE_LMP_BY_SETTLEMENT_POINT_RTID,
+            date=date,
+            extension="csv",
+            published_before=end,
+            published_after=date,
+            verbose=verbose,
+        )
+        df = self.read_docs(docs, parse=False, verbose=verbose)
+        return self._handle_indicative_lmp_by_settlement_point(df)
+
+    def _handle_indicative_lmp_by_settlement_point(self, df: pd.DataFrame):
+        columns_to_rename = {
+            "IntervalId": "Interval Id",
+            "IntervalEnding": "Interval End",
+            "RTDTimestamp": "RTD Timestamp",
+            "SettlementPoint": "Location",
+            "SettlementPointType": "Location Type",
+            "LMP": "LMP",
+        }
+        df.rename(columns=columns_to_rename, inplace=True)
+        assert set(df["RepeatedHourFlag"].unique()).issubset({"Y", "N"})
+        assert set(df["IntervalRepeatedHourFlag"].unique()).issubset({"Y", "N"})
+
+        df["RTD Timestamp"] = pd.to_datetime(df["RTD Timestamp"]).dt.tz_localize(
+            self.default_timezone,
+            ambiguous=df["RepeatedHourFlag"] == "N",
+            nonexistent="shift_forward",
+        )
+        df["Interval End"] = pd.to_datetime(df["Interval End"]).dt.tz_localize(
+            self.default_timezone,
+            ambiguous=df["IntervalRepeatedHourFlag"] == "N",
+            nonexistent="shift_forward",
+        )
+
+        df["Interval Start"] = df["Interval End"] - pd.Timedelta(minutes=5)
+        df = df.sort_values("Interval Start").reset_index(drop=True)
+        return df[
+            [
```

Comment:
i'm thinking about the column order. i think this makes more sense and also matches the ercot source 

```
 [
                "RTD Timestamp",
                "Interval Start",
                "Interval End",
                "Interval Id",
                "Location",
                "Location Type",
                "LMP",
            ]
```